adapters/amx/amx.go

```diff
@@ -0,0 +1,174 @@
+package amx
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+	"strings"
+)
+
+const vastImpressionFormat = "<Impression><![CDATA[%s]></Impression>"
+const vastSearchPoint = "</Impression>"
+
+// AmxAdapter is the AMX bid adapter
+type AmxAdapter struct {
+	endpoint string
+}
+
+// NewAmxBidder creates an AmxAdapter
+func NewAmxBidder(endpoint string) *AmxAdapter {
+	return &AmxAdapter{endpoint: endpoint}
+}
+
+type amxExt struct {
+	Bidder amxParams `json:"bidder"`
+}
+
+type amxParams struct {
+	PublisherID string `json:"tagId,omitempty"`
+}
+
+func getPublisherID(imps []openrtb.Imp) *string {
+	for _, imp := range imps {
+		paramsSource := (*json.RawMessage)(&imp.Ext)
+		var params amxExt
+		if err := json.Unmarshal(*paramsSource, &params); err == nil {
+			if params.Bidder.PublisherID != "" {
+				return &params.Bidder.PublisherID
+			}
+		} else {
+			fmt.Printf("unable to decode imp.ext.bidder: %v", err)
```

Comment:
There is no consequence to this print statement, neither in the response or the in the log. The chance of this happening is also likely 0% due to preprocessing by the "core" logic. I recommend removing the print statement since the publisher id is optional for your adapter.